src/main/java/domain/Point.java

```diff
@@ -0,0 +1,69 @@
+package domain;
+
+import exception.UnavailabeCoordinateException;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+
+public class Point {
+    private static final int X_COORDINATE_POSITION = 0;
+    private static final int Y_COORDINATE_POSITION = 1;
+    private static final int NECESSARY_COORDINATES_SIZE = 2;
+    private static final String COORDINATE_DELIMITER = ",";
+    private Coordinate xCoordinate;
+    private Coordinate yCoordinate;
+
+    public Point(String rawCoordinates) {
+        List<String> coordinates = Arrays.asList(rawCoordinates.split(COORDINATE_DELIMITER));
+        checkCoordinatesQuantity(coordinates);
+        try {
+            this.xCoordinate = Coordinates.retrieveValue(Integer.parseInt(coordinates.get(X_COORDINATE_POSITION)));
+            this.yCoordinate = Coordinates.retrieveValue(Integer.parseInt(coordinates.get(Y_COORDINATE_POSITION)));
+        } catch (NumberFormatException e) {
+            throw new UnavailabeCoordinateException();
+        }
+    }
+
+    private void checkCoordinatesQuantity(List<String> coordinates) {
+        if (coordinates.size() != NECESSARY_COORDINATES_SIZE) {
+            throw new UnavailabeCoordinateException();
+        }
+    }
+
+    public double calculateDistance(Point otherPoint) {
+        double xSquared = this.xCoordinate.calculateSquaredDistance(otherPoint.xCoordinate);
```

Comment:
```suggestion
        double xSquared = Math.pow(this.xCoordinate.getValue - otherPoint.xCoordinate.getValue, 2);
```

이렇게 쓰는게 더 깔끔하지 않을까요?
`this.xCoordinate.getValue - otherPoint.xCoordinate.getValue` 이 부분은 메소드로 빼도 괜찮을 것 같은데, 굳이 제곱까지 Coordinate 에서 할 필요는 없는 것 같아요.
처음에 Coordinate 에서 제곱을 하길래 루트를 씌우면 소숫점이 나와서 부정확할까봐 그런건가 했네요.